### PR TITLE
Bug 1881865: AWS machine does not fail on missing userData secret

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Prevent machine from being stuck at a pending state forever, and show aproptiate error on missing userData secret.

### CODE REVIEW

File: pkg/actuators/machine/machine_scope.go [hunk 1]
```diff
@@ -122,13 +122,15 @@ func (s *machineScope) getUserData() ([]byte, error) {
 		Name:      s.providerSpec.UserDataSecret.Name,
 	}
 
-	if err := s.client.Get(s.Context, objKey, userDataSecret); err != nil {
+	if err := s.client.Get(s.Context, objKey, userDataSecret); apierrors.IsNotFound(err) {
+		return nil, machineapierros.InvalidMachineConfiguration("userData secret is not found: %#v", objKey)
```
Comment (JoelSpeed) [inline pkg/actuators/machine/machine_scope.go:126]: We wouldn't typically use `%#v` to print something like this for an error message, it's not particularly user friendly, how about this instead?
```suggestion
		return nil, machineapierros.InvalidMachineConfiguration("userData secret is not found: %s/%s", objKey.Namespace, objKey.Name)
```

**Review by JoelSpeed — COMMENTED** [review]
Are machines currently looping on trying to create if they don't have a valid `userdatasecret`? I wonder if that's actually desirable? Means that the user data secret need not be present before the Machine is created, but it will get created eventually (which IMO is fine)

/hold

I'm wondering if this will break IPI clusters, what if MCO hasn't generated the secret before the Machines are created? Won't this cause all of the Machines to go failed immediately, failing the install?

**Comment by michaelgugino** [comment]
> Are machines currently looping on trying to create if they don't have a valid `userdatasecret`? I wonder if that's actually desirable?

There could be sophisticated users that want to generate a user-data secret on a per-machine basis for one reason or another, they might create a controller to watch machine creations and then stamp out something that matches what's in the machine's spec.

This might especially be true if we go to 1 time keys for the bootstrap configs, we'd need to get a user-data secret per machine.

**Comment by Danil-Grigorev** [comment]
Closing the PR based on Mike's comment. This BZ is a duplicate of another one, which referenced by https://github.com/openshift/machine-api-operator/pull/673. If we decide to go any way, the fix will be generic.

**Pull request closed without merging**